setup.py

```diff
@@ -42,6 +42,10 @@
 # Read in requirements
 requirements = open('requirements.txt').readlines()
 requirements = [r.strip() for r in requirements]
+# install_requires should not be overly specific.
+# Change any == pinned deps to >=.
+# https://packaging.python.org/discussions/install-requires-vs-requirements/
+requirements = [r.replace('==', '>=') for r in requirements]
```

Comment:
Remove this change. Our requirements.txt file should be the literal source of truth. This also causes the code installed on user machines to be broken, e.g. it uses the bad version of sympy.

Reply:
So I suggested this change. I agree that as written it has sympy issues. In general, I'd advocate for never being overly specific on which versions are required for a developer-oriented library. Usually the way python packages can handle this is by pinning things in `requirements.txt` so e.g. CI runs under a consistent environment and users who don't care can `pip install -r requirements.txt` and get a good environment and leaving `install_requires` with a looser set of restrictions so `pip install cirq` has a low probability of upgrading or downgrading an existing install of a package (particularly heavy-weight ones like matplotlib or numpy). 

If you insist on having one set of version specs for these two use cases, each restriction in `requirements.txt` should have a corresponding reason associated with it. If the reason is "there's a bug in the most recent release" then that's good and a future developer knows the condition under which it is safe to change the version spec. If the reason is "well I seem to have pandas 0.25.1 and I can't be bothered to check older or newer versions" ... that's less compelling

Reply:
If we are going to have reasons required for restrictions in `requirements.txt`, seems like this PR isn't useful anymore, because the SymPy stuff was fixed. I can revert if that's the direction we want to go in.